mergin/client.py

```diff
@@ -915,6 +925,58 @@ def pull_project(self, directory):
         pull_project_wait(job)
         return pull_project_finalize(job)
 
+    def sync_project(self, project_dir):
+        """
+        Syncs project by loop with these steps:
+        1. Pull server version
+        2. Get local changes
+        3. Push first change batch
+        Repeat if there are more changes pending.
+        """
+        has_more_changes = True
+        while has_more_changes:
+            self.pull_job = pull_project_async(self, project_dir)
+            pull_project_wait(self.pull_job)
+            pull_project_finalize(self.pull_job)
+
+            changes_handler = ChangesHandler(self, project_dir)
+            changes_batch, has_more_changes = changes_handler.get_change_batch()
+
+            self.push_job = push_project_async(project_dir, changes_batch)
+            push_project_wait(self.push_job)
+            push_project_finalize(self.push_job)
+
+    def sync_project_with_callback(self, project_dir, progress_callback=None, sleep_time=100):
+        """
+        Syncs project while sending push progress info as callback.
+        Sync is done in this loop:
+            Pending changes? -> Pull -> Push change batch -> repeat
+        """
+        has_more_changes = True
+        while has_more_changes:
+            changes_handler = ChangesHandler(self, project_dir)
+
+            self.pull_job = pull_project_async(self, project_dir)
+            if self.pull_job is None:
+                return
+            pull_project_wait(self.pull_job)
+            pull_project_finalize(self.pull_job)
+
+            changes_batch, has_more_changes = changes_handler.get_change_batch()
+
+            self.push_job = push_project_async(self, project_dir, changes_batch)
+            if not self.push_job:
+                return
```

Comment:
is it correct to have the early return there? I guess if there are multiple batches, in theory there could be a single batch with just file removals (thus not uploading any data), and later some batches with data - and this early return would skip the other batches